### PR TITLE
Local Token Decoding

### PR DESCRIPTION
**Pull request by chmoder**
Hi, we found this library when integrating keycloak and django.  It is a simple, easy to use library, thanks for sharing it!  One thing we noticed is that there are three HTTP requests from the middleware to keycloak.

Two to `/token/introspect`
One to `/openid-connect/userinfo`

The total time for all three requests is around **.80** seconds.

- adds django caching
- adds setting to use local token decoding instead of keycloak endpoints (default to use http interface)
- adds conditonal checks in `introspect`, `is_token_active`, `roles_from_token`,  and `userinfo` to use local decoding or http interface
- add jwks endpoint and method (with caching)
- adds token decode method

### CODE REVIEW

**Comment by marcelo225** [comment]
@chmoder, nice contributing!
I'll check this out soon.
Thanks for helping us!

**Comment by marcelo225** [comment]
@chmoder, could you change at [setup.py file](https://github.com/InsuranceToolkits/django-keycloak-auth/blob/master/setup.py#L11) the new version 0.9.9?

File: setup.py [hunk 1]
```diff
@@ -14,6 +14,8 @@
     # Project uses reStructuredText, so ensure that the docutils get
     # installed or upgraded on the target machine
     install_requires=[
+        "cryptography",
```
Comment (marcelo225) [inline setup.py]: @chmoder, Could you freeze these newly added libraries? I am afraid there might be some problems when someone uses this library. What do you think?

File: setup.py [hunk 2]
```diff
@@ -14,6 +14,8 @@
...
+        "PyJWT",
```
Comment (marcelo225) [inline setup.py]: This could be changed on this line as well

**Comment by marcelo225** [comment]
@chmoder, I've updated a #28 PR into the master branch recently. Could you merge it into your branch?

File: setup.py (hunk 1 above)
Comment (chmoder) [reply setup.py]: Great suggestions!  I pinned the version of pyJWT to the latest or anything that comes after.  And pyJWT has required cryptography>=3.4.0 for ~2 years now so I pinned ours to that dependency as well.

**Comment by chmoder** [comment]
> @chmoder, I've updated a #28 PR into the master branch recently. Could you merge it into your branch?

@marcelo225, I merged django-keycloak-auth master branch into this fork.  All your suggestions should be contributed as well.

I [added caching](https://github.com/InsuranceToolkits/django-keycloak-auth/blob/master/django_keycloak_auth/settings.py#L170) for the `jwks` endpoint.  Today I made the TTL for that a setting and updated documentation.  

Let me know what you think, the idea comes from [min-time-between-jwks-requests](https://www.keycloak.org/docs/latest/securing_apps/) and related documentation.  It may even be helpful for the `well_known_endpoint`.

**Comment by marcelo225** [comment]
@chmoder , The new lib version 0.9.9 is now available. 
Thanks for contributing to it!

**Pull request merged**